### PR TITLE
Enable users to give tx fee to block proposers

### PR DESCRIPTION
**Pull request by h0n9**
- rename 'distributeReward' to 'distributeIncentive'
- revise definition of 'incentive' = 'reward' + 'fee'
- change 'numTxs' to 'numDeliveredTxs' in order to give incentive for ONLY verified txs

### CODE REVIEW

File: amo/amo.go [hunk 1]
```diff
@@ -287,17 +289,37 @@ func (app *AMOApp) DeliverTx(req abci.RequestDeliverTx) abci.ResponseDeliverTx {
 	tags := []tm.KVPair{
 		{Key: []byte("tx.type"), Value: []byte(t.GetType())},
 		{Key: []byte("tx.sender"), Value: []byte(t.GetSender().String())},
+		{Key: []byte("tx.fee"), Value: []byte(t.GetFee().String())},
```
Comment (Lbird) [inline amo/amo.go]: Remove this. We don't need to index for fee.

File: amo/tx/tx.go [hunk 2]
```diff
@@ -54,6 +56,7 @@ type TxBase struct {
 	Sender    crypto.Address  `json:"sender"`
 	Nonce     tm.HexBytes     `json:"nonce"`
 	Payload   json.RawMessage `json:"payload"` // TODO: change to txparam
+	Fee       types.Currency  `json:"fee"`
```
Comment (Lbird) [inline amo/tx/tx.go:58]: After Nonce.

File: amo/tx/tx.go [hunk 3]
```diff
@@ -62,6 +65,7 @@ type TxToSign struct {
 	Sender    crypto.Address  `json:"sender"`
 	Nonce     tm.HexBytes     `json:"nonce"`
 	Payload   json.RawMessage `json:"payload"`
+	Fee       types.Currency  `json:"fee"`
```
Comment (Lbird) [inline amo/tx/tx.go:67]: After Nonce.

**Review by Lbird — CHANGES_REQUESTED** [review]

**Comment by h0n9** [comment]
Fix error on circleCI by changing the order of tx structrue's properties
related to: https://github.com/amolabs/amo-client-go/commit/d844be82eb3ec5e3204367ae649f11296ba7610b

**Review by Lbird — APPROVED** [review]

**Pull request merged**